### PR TITLE
ci: batch-land gate hardening + cross-platform fixes via the M5 runner

### PR DESCRIPTION
**Pull request by danielraffel**
Bundles four code-proven PRs into one to land in a **single macOS CI run on the idle M5/blackbook runner**, instead of one run each overflowing to saturated github-hosted (the runner-capacity bottleneck diagnosed this session). The macOS leg is retargeted to the M5 via `operator-override`; the github-hosted advisory legs already passed on each.

Supersedes (close on merge):
- #3388 — `test(render/gpu)`: contention-robust GPU-perf sentinel (median + env-gated strict threshold)
- #3361 — `fix(build)`: link system ICU on Linux + setup.sh libdrm probe
- #3394 — `ci`: surface self-hosted ctest failures in the job summary + artifact
- #3404 — `chore(deps)`: bump Shipyard pin v0.59.0 → v0.67.0 (rescue re-resolve, --adopt-head, diagnostics)

Combining also resolves the #3361↔#3388 ordering (the GPU-test fix and the Linux fix are now in the same branch). Each leg's CI was green individually; this is purely a CI-run-count + capacity optimization.

### CODE REVIEW

File: test/test_gpu_compute.cpp [hunk 1]
```diff
@@ -266,32 +268,57 @@ TEST_CASE("GpuCompute magnitude hot loop reuses pool buffers",
...
-    for (int i = 0; i < kIters; ++i) {
-        REQUIRE(compute->compute_magnitude(
-            complex_pairs.data(), magnitudes.data(), N));
+    // Measure several batches and take the MEDIAN per-call time. CI runners are
+    // shared; a single averaged run can be inflated by a transient host-load
+    // spike (a concurrent build, another VM's GPU work), which previously made
+    // this assertion flake on busy hosts. The median across batches resists a
+    // one-off spike without masking a real, sustained regression.
+    constexpr int kBatches = 5;
+    constexpr int kItersPerBatch = 1000;
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply test/test_gpu_compute.cpp:277]: Addressed in #3413: reduced the hot-loop from 5×1000 to 5×100 submits. The root issue was that total synchronous submits are bounded by `ctest --timeout 120`, so 5000 submits gave a ~24 ms/call wall-clock ceiling — *below* the 100 ms/call runaway sentinel, making the sentinel unreachable (a real leak would time the test out at the ceiling instead of tripping the sentinel). 500 submits gives a ~240 ms/call ceiling, comfortably above the sentinel, with an explanatory comment tying the iter count to the ctest timeout. #3413 also lands the durable routing fix (#3299) so this test stops landing on saturated overflow runners in the first place.
